Source/Order.swift

```diff
@@ -121,5 +114,72 @@ Represents an order placed by the user to be tracked using `ButtonMerchant.track
         case lineItems = "line_items"
         case customer
     }
+    
+    /**
+     Represents a customer in the order.
+     */
+    public class Customer: NSObject, Codable {
+        
+        /**
+         The id for the transacting customer in your system (required).
+         */
+        let id: String
+        
+        /**
+         The email of the transacting customer.
+         */
+        let email: String?
+        
+        /**
+         The SHA-256 hash of the transacting customer’s lowercase email, as a 64-character hex string.
+         The value of the e-mail address must be converted to lowercase before computing the hash.
+         The hash itself may use uppercase or lowercase hex characters.
+         */
+        let emailSha256: String?
+        
+        /**
+         The customer’s IDFA.
+         */
+        let advertisingId: String?
+        
+        /**
+         Initializes a customer object with the passed parameters.
+         
+         - Parameters:
+         - id: The id for the transacting customer in your system (required).
+         - email: The email of the transacting customer.
+         - advertisingId: The customer’s IDFA.
+         */
+        @objc public init(id: String, email: String?, advertisingId: String? = nil) {
+            self.id = id
+            self.email = email?.lowercased()
```

Comment:
We need to do the has if they set this field—or are we doing that elsewhere? cc @Jon6193